attributes-next.md

```diff
@@ -0,0 +1,252 @@
+- Deprecation warnings will be emitted for edge cases where the APIs differ
+- Ater fixing the deprecation warnings, autofixing from the old APIs to the new APIs should be possible
+
+Three Separated Namespaces:
+
+- Ohai attributes
+- Default and override attributes
+- Normal/storage attributes
+
+Deprecated Behavior:
+
+- Array deep merging is removed (must be manually replaced with hash-key merging)
+- Deep merging of normal values with default and override is removed (users who were incorrectly using normal
+  as a third precedence level will need to manually fix)
+- Deep merging of automatic values with default/normal/override is removed (this is likely just namespace
+  collision mistakes and stright-up bugs?)
+- The force_default/force_override levels are now the standard behavior of setting attributes in recipe code, the
+  "force" behavior becomes the standard behavior.
+- Environment files consistently have a higher priority / later merge order than roles.
+
+Merged Precedence Levels (Current API):
+
+| level | lowest  |  | | highest |
+| ---- | ------ | ------ | ---- | ---- |
+| default       | attribute | recipe | environment | role |
+| force_default | attribute | recipe | | |
+| normal | attribute | recipe | | |
+| override | attribute | recipe | role | environment |
+| force_override | attribute | recipe | | |
+| automatic| node |
+
+Merged Precedence Levels (New API):
+
+| level | lowest  |  | | highest |
+| ---- | ------ | ------ | ---- | ---- |
+| default       | attribute | role | environment | recipe
+| override | attribute | role | environment | recipe
+
+- automatic attributes are no longer merged so are not shown in the above table (they still exist)
+- `force_default/force_override` becomes the way attributes are applied in recipes
+
+## New Syntax for Ohai Attributes
+
+A new sugar for ohai attributes will be introduced:
+
+```ruby
+ohai("platform")
+ohai("cloud_v2", "public_ipv4_addrs", 0)
+```
+
+This will literally be sugar for the `node.automatic.read("platform")` API that has already been
+shipped.
+
+Write access to ohai data will continue through the existing `node.automatic` API.  In a future RFC
+that will be replaced.
+
+## New Syntax for Merged Default and Override Attributes
+
+```ruby
+attr("foo", "bar", "baz")
+```
+
+This is a reader-only that only reads values merged from default and override levels, with hash-only merging.
+
+It is not intended to be identical to the existing `node["foo"]["bar"]["baz"]` although once deprecation warnings
+against the node object are fixed (see below) they should converge to the same semantics.
+
+The order of merging of the hashes will be:
+
+- default (attribute files)
+- role_default
+- env_default
+- force_default (recipe files)
+- override (attribute files)
+- role_override
+- env_override
+- force_override (recipe files)
+
+Note (below) that attributes written in recipe files will go into the force_default/force_override hashes and will not
```

Comment:
OK so you're calling out here the different storage facilities, but then you need to support force_override! (force_default! and force_default do the same thing since they are the lowest precedent)